File: day:20/mergeSortedArrays.py
class Solution:
    def addToNums1(self, nums1, x):
        nums1.append(x)
        for i in range(len(nums1) - 1, -1, -1):
            if i == 0:
                break
            if nums1[i] < nums1[i - 1]:
                [nums1[i], nums1[i - 1]] = [nums1[i - 1], nums1[i]]
                continue
            else:
                break
            
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        # base cases
        if len(nums1) == 0 or m == 0:
            if m == 0:
                while len(nums1) > 0:
                    nums1.pop()
            for i in nums2:
                nums1.append(i)
            return
        if len(nums2) == 0 or n == 0:
            if n == 0:
                while len(nums2) > 0:
                    nums2.pop()
            for i in nums1:
                nums2.append(i)
            return
        
        while len(nums1) > m:
            nums1.pop()
        while len(nums2) > n:
            nums2.pop()
        if nums2[0] >= nums1[len(nums1) - 1]:
            for x in nums2:
                nums1.append(x)
            return nums1
        for x in nums2:
            self.addToNums1(nums1, x)

        return nums1
        

File: day:20/mergeSortedArrays.js
function addToNums1(nums1, x) {
    nums1.push(x);
    for(let i = nums1.length - 1; i >= 0; i--) {
        if(i === 0) break;
        if(nums1[i] < nums1[i - 1]) {
            [nums1[i], nums1[i-1]] = [nums1[i-1],nums1[i]];
            continue;
        }
        else {
            break;
        }
    }
}
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    while(nums1.length > m) {
        nums1.pop();
    }
    while(nums2.length > n) {
        nums2.pop();
    }
    // base case
    if(nums2[0] >= nums1[nums1.length - 1]) {
        for(let x of nums2) {
            nums1.push(x);
        }
        return nums1;
    }

    for(let x of nums2) {
        addToNums1(nums1, x);
    }
    return nums1;
};
